src/data-fetching/lib/getEnvVariables.ts

```diff
@@ -27,18 +27,21 @@ function decodeHttpEnv(header: string): Record<string, any> {
     return decodeJson(header);
 }
 
-export function getEnvVariables(req?: IncomingMessage): PrezlyEnv {
+export function getEnvVariables<EnvType extends PrezlyEnv = PrezlyEnv>(
```

Comment:
Maybe it could look like this? 🤔  So we at least show that we're returning both the `EnvType` and `ProcessEnv`.
With your implementation it seems like we're only returning `EnvType` which is not entirely correct.

```typescript
export function getEnvVariables<EnvType extends PrezlyEnv = PrezlyEnv>(
    req?: IncomingMessage,
): NodeJS.ProcessEnv & EnvType {
    if (typeof window !== 'undefined') {
        throw new Error('"getEnvVariables" should only be used on back-end side.');
    }

    const headerName = (process.env.HTTP_ENV_HEADER || '').toLowerCase();

    if (headerName && req) {
        const header = req.headers[headerName] as string | undefined;
        const httpEnv = decodeHttpEnv(header || '');
        return { ...process.env, ...httpEnv } as NodeJS.ProcessEnv & EnvType;
    }

    return { ...process.env } as NodeJS.ProcessEnv & EnvType;
}

Reply:
Yes, that would make more sense! Thanks for looking into this :)

Reply:
I was thinking about adding a check whether the HTTP env header exists (and throwing error if it doesn't) but that doesn't work locally.